axelrod/fingerprint.py

```diff
@@ -379,3 +382,181 @@ def plot(self, cmap: str = 'seismic', interpolation: str = 'none',
+
+        Returns
+        ----------
+        self.data : np.array
+            A numpy array containing the mean cooperation rate against each
+            opponent in each turn. The ith row corresponds to the ith opponent
+            and the jth column the jth turn.
+        """
+
+        if isinstance(self.strategy, axl.Player):
+            players = [self.strategy] + self.opponents
+        else:
+            players = [self.strategy()] + self.opponents
+
+        temp_file_descriptor = None
+        if filename is None:
+            temp_file_descriptor, filename = mkstemp()
+
+        edges = [(0, k + 1) for k in range(len(self.opponents))]
+        tournament = axl.Tournament(players=players,
+                                    edges=edges, turns=turns, noise=noise,
+                                    repetitions=repetitions)
+        tournament.play(filename=filename, build_results=False,
+                        progress_bar=progress_bar, processes=processes)
+
+        self.data = self.analyse_cooperation_ratio(filename)
+
+        if temp_file_descriptor is not None:
+            os.close(temp_file_descriptor)
+            os.remove(filename)
+
+        return self.data
+
+    @staticmethod
+    def analyse_cooperation_ratio(filename):
+        """Generates the data used from the tournament
+
+        Return an M by N array where M is the number of opponents and N is the
+        number of turns.
+
+        Parameters
+        ----------
+        filename : str
+            The filename of the interactions
+
+        Returns
+        ----------
+        self.data : np.array
+            A numpy array containing the mean cooperation rate against each
+            opponent in each turn. The ith row corresponds to the ith opponent
+            and the jth column the jth turn.
+        """
+        did_c = np.vectorize(lambda action: int(action == 'C'))
+
+        cooperation_rates = {}
+        with open(filename, "r") as f:
+            reader = csv.reader(f)
+            for row in reader:
+                opponent_index, player_history = int(row[1]), list(row[4])
+                if opponent_index in cooperation_rates:
+                    cooperation_rates[opponent_index].append(did_c(player_history))
+                else:
+                    cooperation_rates[opponent_index] = [did_c(player_history)]
+
+        for index, rates in cooperation_rates.items():
+            cooperation_rates[index] = np.mean(rates, axis=0)
+
+        return np.array([cooperation_rates[index]
+                         for index in cooperation_rates])
+
+    def plot(self, cmap: str = 'viridis', interpolation: str = 'none',
+             title: str = None, colorbar: bool = True, labels: bool = True,
+             display_names: bool = False) -> plt.Figure:
+        """Plot the results of the spatial tournament.
+
+        Parameters
+        ----------
+        cmap : str, optional
```

Comment:
The above docstrings use `integer` and `string` -- can we chose either the full word or the python functions `int` and `str` (my preference is the latter) throughout?